Properties sanitizer

marandaneto opened the pull request:
## :bulb: Motivation and Context
Some SDKs have a `sanitize_properties` method, others `property_denylist`.
This is similar to `sanitize_properties` but as a callback that is run before the event is cached or sent over the wire. 
The user can redact, remove, or add any properties.
The downside is that the user can mess up with properties that may cause a bug on the backend or something.

## :green_heart: How did you test it?


## :pencil: Checklist
<!--- Put an `x` in the boxes that apply -->

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] I updated the docs if needed.
- [ ] No breaking change or entry added to the changelog.

benjackwhite reviewed (approved):
Personally I like it. I think sanitize properties is the right terminology (certainly as good as any other)

marandaneto commented:
> Personally I like it. I think sanitize properties is the right terminology (certainly as good as any other)

Alright, added tests and updated the usage markdown, will cut a version :)

marandaneto commented:
For context, there was a workaround before:
`PostHog.capture("testEvent", properties = mapOf("\$device_name" to "removed"))`

This would work for manually captured events, but it'd not work for automatically captured events nor events that don't contain the `properties` param eg `identify`.
